assets/src/components/vehiclePropertiesPanel.tsx

```diff
@@ -10,6 +10,66 @@ interface Props {
   selectedVehicleRoute?: Route
 }
 
+const ScheduleAdherenceStatusIcon = ({
+  vehicle: { scheduleAdherenceStatus },
+}: {
+  vehicle: Vehicle
+}) => (
+  <div
+    className={`m-vehicle-properties-panel__schedule-adherence-status-icon ${scheduleAdherenceStatus}`}
+  >
+    <svg width="10" height="10">
+      <circle cx="5" cy="5" r="5" />
+    </svg>
+  </div>
+)
+
+const ScheduleAdherenceStatusString = ({
+  vehicle: { scheduleAdherenceStatus },
+}: {
+  vehicle: Vehicle
+}) => (
+  <div className="m-vehicle-properties-panel__schedule-adherence-status-string">
+    {scheduleAdherenceStatus}
+  </div>
+)
+
+const minutes = (seconds: number): number => Math.floor(seconds / 60)
+
+const earlyOrLateLabel = ({
+  scheduleAdherenceSecs,
+  scheduleAdherenceStatus,
+}: Vehicle): string =>
+  `(${minutes(scheduleAdherenceSecs)} min ${scheduleAdherenceStatus})`
+
+const scheduleAdherenceLabelString = (vehicle: Vehicle): string => {
+  const { scheduleAdherenceStatus } = vehicle
+  if (
+    scheduleAdherenceStatus === "early" ||
+    scheduleAdherenceStatus === "late"
+  ) {
+    return earlyOrLateLabel(vehicle)
+  } else {
+    return ""
+  }
+}
+
+const ScheduleAdherenceLabel = ({ vehicle }: { vehicle: Vehicle }) => (
+  <div className="m-vehicle-properties-panel__schedule-adherencbe-lael">
```

Comment:
typo in the class name? but this class isn't used anyway, so you can probably get rid of this layer of `<div>`.